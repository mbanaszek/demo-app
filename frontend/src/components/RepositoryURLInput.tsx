import React, { FunctionComponent } from "react";
import { Form } from "react-bootstrap";

interface RepositoryURLInputProps {
    repositoryURL: string;
    onRepositoryURLUpdate: (repositoryURL: string) => void;
}

export const RepositoryURLInput: FunctionComponent<RepositoryURLInputProps> = ({
    repositoryURL,
    onRepositoryURLUpdate
}: RepositoryURLInputProps): JSX.Element => {
    return (
        <div>
            <Form.Group controlId="repositoryURL">
                <input
                    type="text"
                    className={"form-control"}
                    placeholder="e.g. 'https://github.com/stoplightio/prism' or 'stoplightio/prism'"
                    value={repositoryURL}
                    onChange={(event) => onRepositoryURLUpdate(event.target.value)}
                />
            </Form.Group>
        </div>
    );
};