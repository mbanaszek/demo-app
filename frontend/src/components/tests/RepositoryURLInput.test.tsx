import React from "react";
import TestRenderer from "react-test-renderer";

import {RepositoryURLInput} from "../RepositoryURLInput";


describe("Render repository URL field", () => {

    test("Empty field.", () => {
        // When
        const component = TestRenderer.create(
            <RepositoryURLInput repositoryURL={''} onRepositoryURLUpdate={()=>{}} />
        );

        // Then
        expect(component.toJSON()).toMatchSnapshot();
    });

    test("With value.", () => {
        // When
        const component = TestRenderer.create(
            <RepositoryURLInput repositoryURL={'https://github.com/mbanaszek/github-leaderboard'} onRepositoryURLUpdate={()=>{}} />
        );

        // Then
        expect(component.toJSON()).toMatchSnapshot();
    });
});
