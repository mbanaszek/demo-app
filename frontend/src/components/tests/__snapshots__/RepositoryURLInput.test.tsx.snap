// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render repository URL field Empty field. 1`] = `
<div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="e.g. 'https://github.com/stoplightio/prism' or 'stoplightio/prism'"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Render repository URL field With value. 1`] = `
<div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="e.g. 'https://github.com/stoplightio/prism' or 'stoplightio/prism'"
      type="text"
      value="https://github.com/mbanaszek/github-leaderboard"
    />
  </div>
</div>
`;
